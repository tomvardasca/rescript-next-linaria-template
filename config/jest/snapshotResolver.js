const path = require('path');

module.exports = {
    resolveSnapshotPath: (testPath) => path.join(
        path.join(path.dirname(testPath).replace('/lib/js', '')),
        path.basename(testPath) + '.snap',
      ),

    resolveTestPath: (snapshotPath) => path.resolve(
        '/lib/js/' + path.dirname(snapshotPath),
        path.basename(snapshotPath).replace('.snap', ''),
      ),

    testPathForConsistencyCheck: path.posix.join(
      '/lib/js/',
      'src',
      'example_test.js',
    ),
  }
