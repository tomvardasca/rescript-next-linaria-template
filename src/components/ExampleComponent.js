// Generated by ReScript, PLEASE EDIT WITH CARE

import * as Curry from "bs-platform/lib/es6/curry.js";
import * as React from "react";

import { css } from "@linaria/core"
;

var button = (css`
    background-color: light-blue;
  `);

var Styles = {
  button: button
};

function ExampleComponent(Props) {
  var match = React.useState(function () {
        return 0;
      });
  var setCount = match[1];
  return React.createElement("div", undefined, React.createElement("button", {
                  className: button,
                  onClick: (function (param) {
                      return Curry._1(setCount, (function (c) {
                                    return c + 1 | 0;
                                  }));
                    })
                }, "Click me"), "Count: " + match[0]);
}

var make = ExampleComponent;

export {
  Styles ,
  make ,
  
}
/*  Not a pure module */
