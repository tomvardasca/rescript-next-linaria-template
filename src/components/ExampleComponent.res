module Styles = %css(
  let button = css`
    background-color: light-blue;
  `
)

@react.component
let make = () => {
  let (count, setCount) = React.useState(_ => 0);
  <div>
    <button className=Styles.button onClick={(_) => setCount((c) => c+1)} >{ "Click me" -> React.string}</button>
    {j`Count: $count` -> React.string} 
  </div>
}