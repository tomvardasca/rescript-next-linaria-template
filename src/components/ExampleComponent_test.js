// Generated by ReScript, PLEASE EDIT WITH CARE

import * as Jest from "@glennsl/bs-jest/src/jest.js";
import * as Curry from "bs-platform/lib/es6/curry.js";
import * as React from "react";
import * as JestDom from "bs-jest-dom/src/JestDom.js";
import * as ExampleComponent from "./ExampleComponent.js";
import * as ReactTestingLibrary from "bs-react-testing-library/src/ReactTestingLibrary.js";

Jest.test("Component renders hook", (function (param) {
        var result = ReactTestingLibrary.render(undefined, undefined, undefined, undefined, undefined, React.createElement(ExampleComponent.make, {}));
        var partial_arg_1 = /Count: 0/;
        var partial_arg = {
          NAME: "RegExp",
          VAL: partial_arg_1
        };
        var arg = function (param, param$1) {
          return ReactTestingLibrary.getByText(partial_arg, param, param$1);
        };
        return JestDom.toBeInTheDocument(expect(Curry._2(arg, undefined, result)));
      }));

Jest.test("Component renders hook click", (function (param) {
        var result = ReactTestingLibrary.render(undefined, undefined, undefined, undefined, undefined, React.createElement(ExampleComponent.make, {}));
        ReactTestingLibrary.act(function (param) {
              var arg = ReactTestingLibrary.FireEvent.click;
              var partial_arg_1 = /Click me/;
              var partial_arg = {
                NAME: "RegExp",
                VAL: partial_arg_1
              };
              var arg$1 = function (param, param$1) {
                return ReactTestingLibrary.getByText(partial_arg, param, param$1);
              };
              Curry._2(arg, undefined, Curry._2(arg$1, undefined, result));
              
            });
        var partial_arg_1 = /Count: 1/;
        var partial_arg = {
          NAME: "RegExp",
          VAL: partial_arg_1
        };
        var arg = function (param, param$1) {
          return ReactTestingLibrary.getByText(partial_arg, param, param$1);
        };
        return JestDom.toBeInTheDocument(expect(Curry._2(arg, undefined, result)));
      }));

export {
  
}
/*  Not a pure module */
