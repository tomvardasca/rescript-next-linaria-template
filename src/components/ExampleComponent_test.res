open Jest;
open ReactTestingLibrary;
open JestDom;

test("Component renders hook", () => {
   let result = <ExampleComponent /> |> render;

    result |> getByText(~matcher=#RegExp(%re("/Count: 0/"))) |> expect |> toBeInTheDocument
});

test("Component renders hook click", () => {
   let result = <ExampleComponent /> |> render;

    act(() =>
      result |> getByText(~matcher=#RegExp(%re("/Click me/"))) |> FireEvent.click |> ignore
    );

    result |> getByText(~matcher=#RegExp(%re("/Count: 1/"))) |> expect |> toBeInTheDocument
});