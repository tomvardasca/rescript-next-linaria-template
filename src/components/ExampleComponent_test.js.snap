// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
<div>
  <div>
    <button
      class="b1j9645y"
    >
      Click me
    </button>
    Count: 0
  </div>
</div>
`;
